modin/core/storage_formats/pandas/aggregations.py

```diff
@@ -0,0 +1,590 @@
+            # they were already replaced by zeroes at the beginning of the 'map' phase.
+            col_vals = np.resize(raw_df[i], raw_df.shape)
+            np.putmask(col_vals, nan_mask, values=0)
+
+            sums[col] = pandas.Series(np.sum(col_vals, axis=1), index=cols, copy=False)
+            sums_of_squares[col] = pandas.Series(
+                np.sum(col_vals**2, axis=1), index=cols, copy=False
+            )
+            count[col] = pandas.Series(
+                nan_mask.shape[1] - np.count_nonzero(nan_mask | nan_mask[i], axis=1),
+                index=cols,
+                copy=False,
+            )
+
+        sums = pandas.concat(sums, axis=1, copy=False)
+        sums_of_squares = pandas.concat(sums_of_squares, axis=1, copy=False)
+        count = pandas.concat(count, axis=1, copy=False)
+
+        return sums, sums_of_squares, count
+
+    @classmethod
+    def reduce(
+        cls, df: pandas.DataFrame, min_periods: int, method: CorrCovBuilder.Method
+    ) -> pandas.DataFrame:
+        """
+        Perform the Reduce phase to compute the corr/cov matrix.
+
+        Parameters
+        ----------
+        df : pandas.DataFrame
+            A dataframe holding aggregations computed for each partition
+            concatenated along the rows axis.
+        min_periods : int
+            Minimum number of observations required per pair of columns to have a valid result.
+        method : CorrCovBuilder.Method
+            Whether to build a correlation or a covariance matrix.
+
+        Returns
+        -------
+        pandas.DataFrame
+            Either correlation or covariance matrix.
+        """
+        if method == CorrCovBuilder.Method.COV:
+            raise NotImplementedError("Computing covariance is not yet implemented.")
+        # The `df` here accumulates the aggregation results retrieved from each row partition
+        # and combined together along the rows axis, so the `df` looks something like this:
+        #   mul  sums  pow2_sums
+        # a .    .     .
+        # b .    .     .            <--- part1 result
+        # c .    .     .
+        # ---------------------------
+        # a .    .     .
+        # b .    .     .            <--- part2 result
+        # c .    .     .
+        # ---------------------------
+        # ...
+        # So to get the total result we have to group on the index and sum the values
+        total_agg = df.groupby(level=0).sum()
+        total_agg = cls._maybe_combine_nan_and_non_nan_aggs(total_agg)
+
+        sum_of_pairwise_mul = total_agg["mul"]
+        sums = total_agg["sum"]
+        sums_of_squares = total_agg["pow2_sum"]
+        count = total_agg["count"]
+
+        cols = sum_of_pairwise_mul.columns
+        # If there were NaNs in the original dataframe then have computed a matrix
+        # of sums/square sums/counts at the Map phase, meaning that we now have multiple
+        # columns in `sums`.
```

Comment:
```suggestion
        # If there are NaNs in the original dataframe, we have computed a matrix
        # of sums/square sums/counts at the Reduce phase 1, meaning that we now have multiple
        # columns in `sums`.
```

Reply:
Corrected (everything except for the `map -> reduce`, [see](https://github.com/modin-project/modin/pull/6193#discussion_r1219790672))